apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue

```diff
@@ -0,0 +1,256 @@
+    try {
+        const fetcher = isPrivate ? SpaceConnector.clientV2.dashboard.privateFolder.create : SpaceConnector.clientV2.dashboard.publicFolder.create;
+        const params: FolderCreateParams = {
+            name: getUniqueFolderName(originFolderName),
+        };
+        if (!isPrivate) {
+            (params as PublicFolderCreateParameters).resource_group = storeState.isAdminMode ? RESOURCE_GROUP.DOMAIN : RESOURCE_GROUP.WORKSPACE;
+        }
+        const createdFolder = await fetcher(params);
+        dashboardMainPageStore.setNewIdList([
+            ...dashboardMainPageState.newIdList,
+            createdFolder.folder_id as string,
+        ]);
+        return createdFolder.folder_id;
+    } catch (e) {
+        return undefined;
+    }
+};
+const createDashboard = async (dashboard: DashboardModel, isPrivate?: boolean, folderId?: string): Promise<boolean> => {
+    try {
+        const _dashboardType = isPrivate ? 'PRIVATE' : 'PUBLIC';
+        const _dashboardWidgets = await listDashboardWidgets(dashboard.dashboard_id);
+        const _sharedLayouts = await getSharedDashboardLayouts(dashboard.layouts, _dashboardWidgets, storeState.costDataSource);
+        let _sharedDashboard: DashboardCreateParams = {
+            name: getUniqueDashboardName(dashboard.name),
+            layouts: _sharedLayouts,
+            options: dashboard.options || {},
+            labels: dashboard.labels || [],
+            tags: { created_by: store.state.user.userId },
+            folder_id: folderId,
+        };
+        if (storeState.isAdminMode) {
+            _sharedDashboard = {
+                ..._sharedDashboard,
+                resource_group: RESOURCE_GROUP.DOMAIN,
+            };
+        } else if (!isPrivate) {
+            _sharedDashboard = {
+                ..._sharedDashboard,
+                resource_group: dashboard?.resource_group || RESOURCE_GROUP.WORKSPACE,
+            };
+        }
+        const createdDashboard = await dashboardStore.createDashboard(_dashboardType, _sharedDashboard);
+        dashboardMainPageStore.setNewIdList([
+            ...dashboardMainPageState.newIdList,
+            createdDashboard.dashboard_id as string,
+        ]);
+        return true;
+    } catch (e) {
+        return false;
+    }
+};
+
+/* Event */
+const handleCloneConfirm = async () => {
+    state.loading = true;
+    const _createDashboardPromises: Promise<boolean>[] = [];
+
+    await Promise.all(dashboardMainPageGetters.selectedTreeData.map(async (item) => {
+        if (item.data.type === 'FOLDER') {
+            const _isPrivate = state.privateMap[item.data.id];
+            const createdFolderId = await createFolder(item.data.name, _isPrivate);
+            item.children.forEach((child) => {
+                _createDashboardPromises.push(createDashboard(child.data, _isPrivate, createdFolderId));
+            });
+        } else {
+            _createDashboardPromises.push(createDashboard(item.data, false));
```

Comment:
issue: I think this flow can cause critical issue.
`createdFolderId` can be undefined when api request has failed.
Even though users wanted to make dashboard under specific folder, in this case, the dashboard would be created on root.